tests/apps/courses/test_cms_plugins_course.py

```diff
@@ -284,3 +285,88 @@ def test_cms_plugins_course_render_variant(self):
         # The new course-glimpse should have the large attribute
         response = self.client.get(url)
         self.assertContains(response, "course-glimpse__large")
+
+    def test_cms_plugins_course_fallback_when_never_published(self):
+        """
+        The course plugin should render in the fallback language when the course
+        page has never been published in the current language.
+        """
+        course = CourseFactory(
+            page_title={"en": "public course", "fr": "cours public"},
+            fill_cover={
+                "original_filename": "cover.jpg",
+                "default_alt_text": "my cover",
+            },
+        )
+        course_page = course.extended_object
+
+        # Create a page to add the plugin to
+        page = create_i18n_page({"en": "A page", "fr": "Une page"})
+        placeholder = page.placeholders.get(slot="maincontent")
+        add_plugin(placeholder, CoursePlugin, "en", **{"page": course_page})
+        add_plugin(placeholder, CoursePlugin, "fr", **{"page": course_page})
+
+        # Publish only the French version of the course
+        course_page.publish("fr")
+
+        # Check the page content in English
+        page.publish("en")
+        url = page.get_absolute_url(language="en")
+        response = self.client.get(url)
+
+        # The course path in french should be in the url the locale in the url
+        # should remain "en"
```

Comment:
```suggestion
        # The course path in french should be in the url.
        # Furthermore the locale in the url should remain "en"
```

Reply:
I added "but" to match what I wrote in other places:
```suggestion
        # The course path in french should be in the url but the locale in the
        # url should remain "en"
```